otx/cli/tools/utils/demo/visualization.py

```diff
@@ -161,3 +165,46 @@ def draw_predictions(task_type: TaskType, predictions: List[Annotation], frame:
     else:
         raise ValueError(f"Unknown task type: {task_type}")
     return frame
+
+
+def get_input_names_list(input_path: Union[str, int], capture):
+    """Lists the filenames of all inputs for demo."""
+
+    # Web camera input
+    if isinstance(input_path, int):
+        return []
+    if "DIR" in capture.get_type():
+        return [f.name for f in Path(input_path).iterdir() if f.is_file()]
```

Comment:
It's for `DirStreamer` right? Then, I think this and that in `DirStreamer` should have same order.
I think one way to ensure that is a making a function to make a list of files in directory and make both of them use it.

Reply:
I'm sorry, I didn't quite understand you.  Can you please elaborate?

Reply:
As far as I understand, function `get_input_names_list` is in charge of providing list of names to save the frames.
And `if "DIR" in capture.get_type():` is for the case where `DirStreamer` is used. The problem is that each frame doesn't have information about which file is used for current frame in that case. So, when saving the frame, get filenames by `[f.name for f in Path(input_path).iterdir() if f.is_file()]` and then using each file name iterating `zip(filenames, saved_frames)`.
What I think as problem is an order of `[f.name for f in Path(input_path).iterdir() if f.is_file()]` in the function and order of `sorted(os.listdir(self.dir))` in `DirStreamer` can be different. So, I think both order of list should be same.
Of course, I may be understanding. then I really appreciate if you let me know :)